src/main/java/com/ibm/stocator/fs/swift/SwiftOutputStream.java

```diff
@@ -72,67 +83,58 @@
+  public SwiftOutputStream(JossAccount account, URL url, final String contentType,
+                           Map<String, String> metadata, SwiftConnectionManager connectionManager)
+          throws IOException {
     mUrl = url;
     totalWritten = 0;
-    HttpURLConnection httpCon = createConnection(account, url, contentType, metadata);
-    try {
-      LOG.debug("Going to obtain output stream for PUT {}", url.toString());
-      mOutputStream  = httpCon.getOutputStream();
-      LOG.debug("Output stream created for PUT {}", url.toString());
-      mHttpCon = httpCon;
-    } catch (ProtocolException e) {
-      LOG.warn("Failed to connect to {}", url.toString());
-      LOG.warn(e.getMessage());
-      LOG.warn("Retry attempt for PUT {}. Re-authenticate", url.toString());
-      account.authenticate();
-      httpCon = createConnection(account, url, contentType, metadata);
-      mOutputStream  = httpCon.getOutputStream();
-      LOG.debug("Second attempt PUT {} successfull. Got output stream", url.toString());
-      mHttpCon = httpCon;
-    } catch (IOException e) {
-      LOG.error(e.getMessage());
-      throw e;
-    }
-  }
-
-  /**
-   * Creates HTTP Connection
-   *
-   * @param account JOSS Account
-   * @param url URL to the object
-   * @param contentType object content type
-   * @param metadata user provided meta data
-   * @return HttpURLConnection if success
-   * @throws IOException if error
-   */
-  private HttpURLConnection createConnection(JossAccount account, URL url, String contentType,
-      Map<String, String> metadata) throws IOException {
-    LOG.debug("Create chunked HTTP connection for PUT {}. Read timeout {}. Streaming chunk {}",
-        url.toString(), READ_TIMEOUT, STREAMING_CHUNK);
-    HttpURLConnection newHttpCon = (HttpURLConnection) url.openConnection();
-    newHttpCon.setDoOutput(true);
-    newHttpCon.setRequestMethod("PUT");
-
-    newHttpCon.addRequestProperty("X-Auth-Token",account.getAuthToken());
-    newHttpCon.addRequestProperty("Content-Type", contentType);
+    mAccount = account;
+    client = connectionManager.createHttpConnection();
+    request = new HttpPut(mUrl.toString());
+    request.addHeader("X-Auth-Token", account.getAuthToken());
     if (metadata != null && !metadata.isEmpty()) {
       for (Map.Entry<String, String> entry : metadata.entrySet()) {
-        newHttpCon.addRequestProperty("X-Object-Meta-" + entry.getKey(), entry.getValue());
+        request.addHeader("X-Object-Meta-" + entry.getKey(), entry.getValue());
+      }
+    }
+
+    PipedOutputStream out = new PipedOutputStream();
+    final PipedInputStream in = new PipedInputStream();
+    out.connect(in);
+    mOutputStream = out;
+    writeThread = new Thread() {
+      public void run() {
+        InputStreamEntity entity = new InputStreamEntity(in, -1);
+        entity.setChunked(true);
+        entity.setContentType(contentType);
+        request.setEntity(entity);
```

Comment:
@djalova i think it's fine to confirm this...as long we know it streams and not buffer internally